icu4c/source/i18n/timezone.cpp

```diff
@@ -382,19 +382,17 @@ createSystemTimeZone(const UnicodeString& id, UErrorCode& ec) {
         return NULL;
     }
     TimeZone* z = 0;
-    UResourceBundle res;
-    ures_initStackObject(&res);
+    StackUResourceBundle res;
     U_DEBUG_TZ_MSG(("pre-err=%s\n", u_errorName(ec)));
-    UResourceBundle *top = openOlsonResource(id, res, ec);
+    UResourceBundle *top = openOlsonResource(id, *res.getAlias(), ec);
```

Comment:
Optional: `*res.getAlias()` looks awkward. We could add a method to StackUResourceBundle that returns a UResourceBundle reference (actually two, a const method that returns a const reference, and a non-const method). Maybe "get()" or "ref()" or "bundle()"?

Reply:
Thanks Markus, I added the two methods to the `StackUResourceBundle` class and force pushed the changes.